### PR TITLE
[SPARK-8675] Executors created by LocalBackend won't get the same classpath as other executor backends 

### PR DESCRIPTION
**Pull request by coderplay**
AFAIK, some spark application always use LocalBackend to do some local initiatives, spark sql is an example. Starting a LocalPoint won't add user classpath into executor.

``` java
  override def start() {
    localEndpoint = SparkEnv.get.rpcEnv.setupEndpoint(
      "LocalBackendEndpoint", new LocalEndpoint(SparkEnv.get.rpcEnv, scheduler, this, totalCores))
  }
```

Thus will cause local executor fail with these scenarios, loading hadoop built-in native libraries, loading other user defined native libraries, loading user jars, reading s3 config from a site.xml file, etc

### CODE REVIEW

**Comment by JoshRosen** [comment]
Jenkins, this is ok to test.

**Comment by andrewor14** [comment]
Looks fine. @coderplay can you fix the scalastyle?

**Comment by coderplay** [comment]
@andrewor14 Sure,  will do. Any existing formatter for eclipse or idea?

**Comment by andrewor14** [comment]
There are some for intellij but Spark has different style requirements than typical Scala applications. Here you can just click into the last test failure and look at the bottom of the test output and fix those manually.

File: core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala [hunk 1]
```diff
@@ -97,10 +100,22 @@ private[spark] class LocalBackend(
...
+  private val userClassPath = getUserClasspath(conf)
+
+  /**
+   * Returns a list of URLs representing the user classpath.
+   *
+   * @param conf Spark configuration.
+   */
+  def getUserClasspath(conf: SparkConf): Seq[URL] = {
+    val userClassPathStr = conf.getOption("spark.executor.extraClassPath")
+    userClassPathStr.map(_.split(File.pathSeparator)).toSeq.flatten.map(new URL(_))
```
Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala]: you might need to add `.map { p => new URL("file:/" + p.stripPrefix("/")) }`

Comment (coderplay) [inline core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala]: it's a solution, but what if user specify a hdfs://xx/yy.jar ?

Comment (coderplay) [inline core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala]: From https://github.com/apache/spark/blob/master/yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala#L202 ,   seems spark doesn't sport non-local classpath

Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/scheduler/local/LocalBackend.scala]: if you're running in local mode it's unlikely your classpath will be from hdfs

**Comment by coderplay** [comment]
```
======================================================================
FAIL: test_time_with_timezone (__main__.SQLTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/SparkPullRequestBuilder/python/pyspark/sql/tests.py", line 716, in test_time_with_timezone
    self.assertEqual(now, now1)
AssertionError: datetime.datetime(2015, 7, 9, 18, 12, 37, 910045) != datetime.datetime(2015, 7, 9, 18, 12, 37, 910044)
```

Seems is caused by commitment c9e2ef52bb54f35a904427389dc492d61f29b018

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
Merging into master, thanks!

**Comment by andrewor14** [comment]
@coderplay can you close this now? I already merged it in master. Thanks for your changes.

**Pull request closed without merging**